### PR TITLE
updadte doc of KlusterletSpec.Namespace

### PR DESCRIPTION
**Pull request by elgnay**
Signed-off-by: Yang Le <yangle@redhat.com>

### CODE REVIEW

File: operator/v1/types.go [hunk 1]
```diff
@@ -247,11 +247,14 @@ type Klusterlet struct {
...
 // KlusterletSpec represents the desired deployment configuration of Klusterlet agent.
 type KlusterletSpec struct {
-	// Namespace is the namespace to deploy the agent.
+	// Namespace is the namespace to deploy the agent on the managed cluster.
 	// The namespace must have a prefix of "open-cluster-management-", and if it is not set,
 	// the namespace of "open-cluster-management-agent" is used to deploy agent.
-	// Note: in Detach mode, this field will be **ignored**, the agent will be deployed to the
-	// namespace with the same name as klusterlet.
+	// In addition, the add-ons are deployed to the namespace of "{Namespace}-addon".
+	// Note: In the Hosted mode, this namespace still exists on the managed cluster to
```
Comment (qiujian16) [inline operator/v1/types.go]: we probably should remove `Note:`. It is not necessary in api doc

**Review by qiujian16 — COMMENTED** [review]
/approve

File: operator/v1/types.go (hunk 1 above)
Comment (elgnay) [reply operator/v1/types.go]: Updated.

**Pull request merged**